hphp/runtime/base/runtime-error.cpp

```diff
@@ -24,7 +24,29 @@
 
 namespace HPHP {
 ///////////////////////////////////////////////////////////////////////////////
+static int64_t g_warning_counter = 0;
+static int64_t g_notice_counter = 0;
+
+bool checkReportLevel(ErrorMode mode) {
+  int errnum = static_cast<int>(mode);
+  if (!g_context->errorNeedsHandling(errnum, true,
```

Comment:
Since we checked need handling here, can we remove   https://github.com/huzhiguang/hiphop-php/blob/patch-13/hphp/runtime/base/runtime-error.cpp#L326 the checking  here? they are duplicated.